modules/noise/noise_texture_2d.cpp

```diff
@@ -194,6 +194,9 @@ Ref<Image> NoiseTexture2D::_modulate_with_gradient(Ref<Image> p_image, Ref<Gradi
 
 void NoiseTexture2D::_update_texture() {
 	bool use_thread = true;
+#ifndef THREADS_ENABLED
+	use_thread = false;
+#endif
```

Comment:
This can be simplified to:

```suggestion
#ifdef THREADS_ENABLED
	bool use_thread = true;
#else
	bool use_thread = false;
#endif
```

This works because the C++ preprocessor makes it so you don't redefine the variable in this scenario.

Reply:
Yes but then why not go a step further and remove the `if (first_time)` and `if (use_thread)`?

I tried to use the `#ifdef` here like a normal `if`, and to let the compiler optimize away the rest.

It starts with `use_thread = true` and then has some checks that can disable threads, such as `if (first_time)` or `#ifdef THREADS_ENABLED`. The fact that one of these is a compile time check shouldn't change the flow IMO.

Reply:
I don't think redefining the variable is a problem. The compiler will definitely produce the same result. I find the current approach in the PR to be simpler. So my preference is to keep it the same